### DISCUSSION TITLE
Stage Bot

### DISCUSSION THREAD

**Discussion by ads231** [eritislami/evobot#1002] [Q&A]
how do i make a discord music stage bot like preo and other bots which stays in vc. I search some codes but when i try to make them they doesn't work. I do coding stuff in repl.com so if anyone have a code then gimme please i m searching from 2 months

**Comment by Takiyo0** [comment]
If stage channel is what you meant, no you can't for now, at least. Stage channel requires discord.js v13, meanwhile this project still using discord.js v12. And if you want to upgrade it to discord v13, it's not that simple as using `npm i discord.js@13`. You need to change almost all files for it to work and neither single line nor single file, since it's a major update and not a minor update.

And for `and other bots which stays in vc`, i'd assume that you need the bot stay in the vc 24h without leaving. You can remove these lines
```
      setTimeout(function () {
        if (queue.connection.dispatcher && message.guild.me.voice.channel) return;
        queue.channel.leave();
        !PRUNING && queue.textChannel.send(i18n.__("play.leaveChannel"));
      }, STAY_TIME * 1000);
``` 
here https://github.com/eritislami/evobot/blob/4681a9fca4ed0225da4ebff94aef31d6757efc4d/include/play.js#L22
You also need the bot to run 24h non-stop. You need VPS for it and you can't use other free host like repl.it, heroku, or glitch. It can't run than 24 hours. But if you really need it for at least running, you can refer to this discussion to host on repl.it https://github.com/eritislami/evobot/discussions/953#discussioncomment-1413432

**Reply by Shaxpartan** [reply]
You can use solonodes.xyz
its 24h not stop
